src/rsa_api/rsa_api.py

```diff
@@ -2499,3 +2542,81 @@ def DEVICE_GetTemperature(self, unit: str = "celsius") -> float:
+        """
+        Stream IQ data to a NumPy array.
+
+        Parameters
+        ----------
+        duration_msec : int
+            Duration of time to record IQ data, in milliseconds.
+        return_status : bool
+            Whether or not to return the IQ capture status message.
+            If False, errors will be raised for buffer overflow and
+            input overrange events.
+
+        Returns
+        -------
+        iq_data : np.ndarray of np.complex64 values
+            IQ data, with each element in the form (I + j*Q)
+        iq_status : str (optional)
+            The status string for the IQ capture, as defined in
+            the documentation for IQSTREAM_StatusParser().
+        """
+        dest = _IQS_OUT_DEST[0]  # Client
+        dtype = _IQS_OUT_DTYPE[0]  # Single
+        buffer_size = 1000000  # Maximum
+
+        # Ensure device is stopped before proceeding
+        self.DEVICE_Stop()
+
+        # Configure IQ Streaming
+        self.IQSTREAM_SetOutputConfiguration(dest, dtype)
+        self.IQSTREAM_SetIQDataBufferSize(buffer_size)
+        buffer_size = self.IQSTREAM_GetIQDataBufferSize()
+        sample_rate_Hz = self.IQSTREAM_GetAcqParameters()[1]
+        iq_samples_requested = int(duration_msec * 1e-3 * sample_rate_Hz)
+        buffer_time_msec = round(buffer_size / sample_rate_Hz * 1e3) * 2
+
+        self.IQSTREAM_ClearAcqStatus()
+        self.DEVICE_PrepareForRun()
+
+        # Initialize data array
+        iqdata = np.empty(iq_samples_requested, dtype=np.complex64)
+        iq_samples_received = 0
+
+        self.DEVICE_Run()
+        self.IQSTREAM_Start()
+
+        while iq_samples_received < iq_samples_requested:
+            # Block while RSA buffer fills
+            while not self.IQSTREAM_WaitForIQDataReady(buffer_time_msec):
+                pass
+            # Then retrieve IQ data (interleaved)
+            iq_block, iq_block_len, iqinfo = self.IQSTREAM_GetIQData(dtype, buffer_size)
+
+            # Deinterleave and store data
+            if iq_samples_received + iq_block_len <= iq_samples_requested:
+                iqdata[iq_samples_received : iq_samples_received + iq_block_len] = (
+                    iq_block[0 : iq_block_len * 2 : 2]
+                    + 1j * iq_block[1 : iq_block_len * 2 : 2]
+                )
+            else:
+                remaining_samples = iq_samples_requested - iq_samples_received
+                iqdata[iq_samples_received:iq_samples_requested] = (
+                    iq_block[0 : remaining_samples * 2 : 2]
+                    + 1j * iq_block[1 : remaining_samples * 2 : 2]
+                )
+            iq_samples_received += iq_block_len
+
+        self.IQSTREAM_Stop()
+        self.DEVICE_Stop()
+
+        assert len(iqdata) == iq_samples_requested
+
+        if return_status:
+            iq_status = self.IQSTREAM_StatusParser(iqinfo, not return_status)
+            return iqdata, iq_status
```

Comment:
Could optionally update method return signature to include iq_status

Reply:
Good call. I made this correction in [af8ca02](https://github.com/NTIA/tekrsa-api-wrap/pull/13/commits/af8ca026899bc2359e396ff741f921c11770574b) for `IQSTREAM_Acquire` as well as `IQSTREAM_Tempfile` and `IQSTREAM_Tempfile_NoConfig` which work the same way.